CauseContract.sol

```diff
@@ -98,42 +105,40 @@ contract CauseContract {
         require(msg.value > 0, "You must send some Ether");
         require(causeState == 1, "This cause has ended, your funds have been returned");
 
-        uint256 gasStart = gasleft();
+        // Use transactionFeeBasisPoints to calculate transactionFee
+        uint256 transactionFee = msg.value * transactionFeeBasisPoints;
+
 
-        uint256 transactionFee = (msg.value*BASIS_POINTS) / 1000; // Transaction fee of 5bps (by default)
-        
-        // (bool success, ) = blockChange.call{value: transactionFee}("");
-        // require(success, "Transfer failed.");
+        // `msg.value - transactionFee` was used twice -> store it in a variable to save gas
+        // Calculate netDonation and use it later for gas optimization
+        uint256 netDonation = msg.value - transactionFee;
 
-        uint256 gasUsed = gasStart - gasleft();
-        uint256 gasPrice = tx.gasprice;
-        uint256 gasFee = gasUsed * gasPrice;
+        // Transfer the transactionFee
+        (bool success, ) = blockChange.call{value: transactionFee}("");
+        require(success, "Transfer failed.");
 
         // update donor proportion
         donorTotals[msg.sender] += msg.value;
 
         // update causeTotal
-        causeTotal += (msg.value - transactionFee);
+        causeTotal += netDonation;
 
-        incoming.push(Transaction(msg.sender, msg.value - transactionFee, block.timestamp, block.number, gasFee, transactionFee));          
+        incoming.push(Transaction(msg.sender, netDonation, block.timestamp, block.number));          
     }
 
     function withdraw(uint256 _amount) public payable onlyAdmin {
         require(address(this).balance > _amount, "Insufficient funds for withdrawal");
-        
-        uint256 gasStart = gasleft();
-        
+       
         causeTotal -= _amount;
```

Comment:
Would you want to remove this line to keep the total donated to the cause? Are you also going to implement the total withdrawn or are these changes for later?

Reply:
Yep just added in an additional variable in the next version